### ISSUE TITLE
The work can be used in finding small exceptional points in images?

### ISSUE THREAD

**Issue by guyiyifeurach** [marco-rudolph/differnet#35]
@marco-rudolph
Hi, Thank you for your nice work!
I want to know that, can your work of differnet or cs-flow can be used to find small exceptional points in X-ray images, which have 3 channels with same pixel data?
First, I want to introduce the backwark. I want to find anomalies of food images which were taken by X-ray tools. These images have 3 channels, but the pixel data in 3 channels are same. And the anomalies in images, such as stainless steel/glass/Al/wire, will block the passage of light produces by X-ray, and produces different texture with normal food.
But the anomalies points in images are very small, because the radius of anomalies materials are very small, some only about 7mm. And I wonder, these small points can be found totally by the differnet or cs-flow?
Thanks for your attention!

**Comment by guyiyifeurach** [comment]
images example:
![image](https://user-images.githubusercontent.com/32059046/137629719-27a2ed00-f3d4-4982-8cdb-641c9d940c77.png)
And red points are anomalies points.

**Comment by marco-rudolph** [comment]
Hello,
thank you.
In principle, this can work, since the feature extractors pre-trained on ImageNet should have seen similar structures. The performance depends on how big the differences in shape and size of the objects within the "normal images" are and how much training data is available. I would also test different feature extractors. Tendentially, I would lean more towards CS-Flow, although here the code has not yet been prepared for publication. Here I guess that the method can better handle the small anomalies, since the feature maps are not averaged.

**Comment by guyiyifeurach** [comment]
In fact, I used object detection to solve the problem, but anomalies are different and can't be predicted,so detecting are not suitable way.In fact, I still insist that unsupervised learning or semi-supervised learning are the right way to solve the problem. And I prepared 2300+ training images with anomalies and labels and locations. Any way to use these images properly?
Can I use different objects images without anomalies as differnet trainable images?This is the question I want to ask after training with one class food image but test value become big suddently.
Then when CS-FLOW will be published?
Thank you for your attention!



---Original---
From: ***@***.***&gt;
Date: Tue, Oct 19, 2021 22:00 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [marco-rudolph/differnet] The work can be used in finding smallexceptional points in images? (Issue #35)




 
Hello,
 thank you.
 In principle, this can work, since the feature extractors pre-trained on ImageNet should have seen similar structures. The performance depends on how big the differences in shape and size of the objects within the "normal images" are and how much training data is available. I would also test different feature extractors. Tendentially, I would lean more towards CS-Flow, although here the code has not yet been prepared for publication. Here I guess that the method can better handle the small anomalies, since the feature maps are not averaged.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by marco-rudolph** [comment]
There is no way for this method to use labeled anomalies in training.
I am not exactly sure how this is meant. So there are different object categories within the training data? Or do the categories differ between the training and test set?
In general, training and testing can be done with different object categories, but the objects being tested on should also be in the training. However, the best performance would probably be achieved if a separate model is used and trained for each object category.
The code for CS-Flow will be released by the end of December at the latest.

**Comment by guyiyifeurach** [comment]
There are different object categories in training data.&nbsp;
Such as SS(stainless steel)/glass/Al, all these small objects will show as 3~5mm black points with different textures.&nbsp;
But I need not classify them, all these points with different textures compared with background should be identified as anomalies. That's my purpose.
As several kinds of anomalies in training data, there are also same kinds of anomalies in testing data.
One model for a kind of anomaly, that's perfect, but we don't have amount of images and time, and model need work for as many as anomalies universally.
Oh, CS-Flow releases so late. In your paper, I speculate the CS-model as Differnet 0~-2 layers and adding postprocessing about calculating max likelihood probability. But how to postprocess wonders me. Would you please show me the roughly postprocess steps or more information? Thank you very much!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "marco-rudolph/differnet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年10月19日(星期二) 晚上10:42
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [marco-rudolph/differnet] The work can be used in finding small exceptional points in images? (Issue #35)





 
There is no way for this method to use labeled anomalies in training.
 I am not exactly sure how this is meant. So there are different object categories within the training data? Or do the categories differ between the training and test set?
 In general, training and testing can be done with different object categories, but the objects being tested on should also be in the training. However, the best performance would probably be achieved if a separate model is used and trained for each object category.
 The code for CS-Flow will be released by the end of December at the latest.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
